skyrim-platform/src/platform_se/skyrim_platform/InventoryApi.cpp

```diff
@@ -239,12 +230,18 @@ JsValue InventoryApi::SetInventory(const JsFunctionArguments& args)
       entry.GetProperty("wornLeft").GetType() != JsValue::Type::Undefined
       ? static_cast<bool>(entry.GetProperty("wornLeft"))
       : false;
-    g_nativeCallRequirements.gameThrQ->AddTask(
-      [pActor, pBoundObject, count]() {
-        pActor->AddObjectToContainer(pBoundObject, nullptr, count, nullptr);
-      });
-    RE::ActorEquipManager* equipManager =
-      RE::ActorEquipManager::GetSingleton();
+    g_nativeCallRequirements.gameThrQ->AddTask([formId, baseId, count]() {
```

Comment:
That's correct

But I think it'd be reasonable to throw exceptions outside of the tasks.

Otherwise, we will not be able to receive this exception on the typescript side (the function would finish successfully, adding a throwing task instead of failing itself)

Let's double-check:
1. throw exceptions in the function itself (not the task)
2. in case of check fails inside a task - do nothing